### PR TITLE
feat: Blitz RPC endpoint to the app directory

### PR DESCRIPTION
**Pull request by siddhsuresh**
### What are the changes and their implications?

Usage:
```ts
// app/api/rpc/[[...blitz]]/route.ts
import {rpcAppHandler} from "@blitzjs/rpc"
import {withBlitzAuth} from "app/blitz-server"

export const {GET, POST, HEAD} = withBlitzAuth(rpcAppHandler())
```

If using only auth and not rpc in the route.ts files
```ts
//app/api/logout/route.ts
export const {POST} = withBlitzAuth({
  POST: async (_request, _params, ctx) => {
    const session = ctx.session
    await session.$revoke()

    return new Response(
      JSON.stringify({
        userId: session.userId,
      }),
      {status: 200},
    )
  },
})
```

## Feature Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)
- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [x] Documentation added/updated (submit PR to [blitzjs.com repo `main` branch](https://github.com/blitz-js/blitzjs.com))

### CODE REVIEW

File: packages/blitz-auth/package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "secure-password": "4.0.0",
-    "typescript": "^4.8.4",
+    "typescript": "5.4.5",
```
Comment (siddhsuresh) [inline packages/blitz-auth/package.json:84]:
```suggestion
    "typescript": "^4.8.4",
```

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

File: packages/blitz-next/src/index-server.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ import {withSuperJsonProps} from "./superjson"
 import {ParsedUrlQuery} from "querystring"
 import {PreviewData} from "next/types"
 import {resolveHref} from "next/dist/client/resolve-href"
-import fs from "fs"
+import fs from "fs-extra"
```
Comment (siddhsuresh) [inline packages/blitz-next/src/index-server.ts:43]:
```suggestion
import fs from "fs"
```

File: packages/blitz-next/src/index-server.ts [hunk 3]
```diff
@@ -243,6 +243,8 @@ export interface BlitzConfig extends NextConfig {
   }
 }
 
+const pwd = __dirname.split(path.sep)
```
Comment (siddhsuresh) [inline packages/blitz-next/src/index-server.ts:246]:
```suggestion
```

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 4]
```diff
@@ -850,7 +974,7 @@ async function createNewSession(
       anonymousSessionToken,
     }
   } else if (global.sessionConfig.method === "essential") {
-    debug("Creating new session")
+    console.log("Creating new session")
```
Comment (siddhsuresh) [inline packages/blitz-auth/src/server/auth-sessions.ts:977]:
```suggestion
    debug("Creating new session")
```

File: packages/blitz-auth/src/server/index.ts [hunk 5]
```diff
@@ -9,8 +9,6 @@ export {
   getSession,
   isLocalhost,
   setPublicDataForUser,
-  setCookie,
   simpleRolesIsAuthorized,
-  getBlitzContext,
```
Comment (siddhsuresh) [inline packages/blitz-auth/src/server/index.ts:14]: Add back

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

File: .changeset/tidy-gorillas-confess.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import {withBlitzAuth} from "app/blitz-server"
+
+// Usage with blitz auth
+export const {GET, POST, HEAD} = withBlitzAuth(rpcAppHandler({}))
+
+// Standalone usage
+export const {GET, POST, HEAD} = rpcAppHandler({})
+```
+
+chore: Update the app directory starter to these latest changes
```
Comment (siddhsuresh) [inline .changeset/tidy-gorillas-confess.md:43]:
```suggestion
chore: Update the app directory starter 
```

**Review by flybayer — COMMENTED** [review]
.

File: .changeset/tidy-gorillas-confess.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+Usage using the new `rpcAppHandler` function
+
+```ts
+// app/api/rpc/[[...blitz]]/route.ts
+import {rpcAppHandler} from "@blitzjs/rpc"
+import {withBlitzAuth} from "app/blitz-server"
+
+// Usage with blitz auth
+export const {GET, POST, HEAD} = withBlitzAuth(rpcAppHandler({}))
```
Comment (siddhsuresh) [inline .changeset/tidy-gorillas-confess.md:37]:
```suggestion
export const {GET, POST, HEAD} = withBlitzAuth(rpcAppHandler())
```

File: .changeset/tidy-gorillas-confess.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+// Standalone usage
+export const {GET, POST, HEAD} = rpcAppHandler({})
```
Comment (siddhsuresh) [inline .changeset/tidy-gorillas-confess.md:40]:
```suggestion
export const {GET, POST, HEAD} = rpcAppHandler()
```

File: .changeset/tidy-gorillas-confess.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+export const handler = withBlitzAuth(async (_request, _params, {session}) => {
+  await session.$revoke()
+  return new Response(
+    JSON.stringify({
+      userId: session.userId,
+    }),
+    {status: 200},
+  )
+})
```
Comment (siddhsuresh) [inline .changeset/tidy-gorillas-confess.md:24]:
```suggestion
export const {POST} = withBlitzAuth({
  POST: async (_request, _params, ctx) => {
    const session = ctx.session
    await session.$revoke()

    return new Response(
      JSON.stringify({
        userId: session.userId,
      }),
      {status: 200},
    )
  },
})
```

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

**Comment by siddhsuresh** [comment]
!preview app-dir-blitz-rpc

**Pull request merged**